src/mrd/ugbio_mrd/mrd_utils.py

```diff
@@ -708,3 +713,408 @@ def generate_synthetic_signatures(
+        df_signatures.join(nunique)
+        .join(
+            filtering_ratio,
+            how="left",
+        )
+        .join(norm_coverage, how="left")
+        .fillna({"filtering_ratio": 1})
+    )
+
+    df_signatures_filt = df_signatures.query(signature_filter_query)
+    return df_signatures, df_signatures_filt
+
+
+def plot_signature_mutation_types(df_signatures_in: pd.DataFrame, signature_filter_query_in: pd.DataFrame):
+    """ "
+    Plot mutation types for a signature dataframe
+    """
+    fig, axs = plt.subplots(1, 2, figsize=(18, 4))
+    fig.suptitle(",".join(df_signatures_in["signature"].unique()), y=1.13)
+    for ax, column, df_plot in zip(
+        axs.flatten(),
+        [
+            "Unfiltered",
+            "Filtered",
+        ],
+        [
+            df_signatures_in,
+            df_signatures_in.query(signature_filter_query_in),
+        ],
+        strict=False,
+    ):
+        x = df_plot["mutation_type"].value_counts(normalize=True).sort_index()
+        all_muts = [
+            "C->A",
+            "C->G",
+            "C->T",
+            "T->A",
+            "T->C",
+            "T->G",
+        ]
+        x = x.reindex(index=all_muts, method="bfill", fill_value=0)
+        tot_mutations = df_plot.shape[0]
+        plt.sca(ax)
+        plt.bar(range(6), x, color=["b", "g", "r", "y", "m", "c"])
+        for px, py in zip(range(6), x, strict=False):
+            plt.text(px, py + 0.01, f"{py:.1%}", ha="center", fontsize=16)
+        plt.ylim(0, ax.get_ylim()[1] + 0.03)
+        plt.yticks([])
+        plt.xticks(range(6), x.index.values, fontsize=20, rotation=90)
+        plt.title(f"{column}, total={tot_mutations:,}", fontsize=28)
+    plt.show()
+
+
+def plot_signature_allele_fractions(df_signatures_in: pd.DataFrame, signature_filter_query_in: pd.DataFrame):
+    """ "
+    Plot allele fraction histograms for a signature dataframe
+    """
+    bins = np.linspace(0, 1, 100)
+    fig, axs = plt.subplots(1, 2, figsize=(18, 4), sharey=True)
+    fig.suptitle(",".join(df_signatures_in["signature"].unique()), y=1.13)
+    for ax, column, df_plot in zip(
+        axs.flatten(),
+        [
+            "Unfiltered",
+            "Filtered",
+        ],
+        [
+            df_signatures_in,
+            df_signatures_in.query(signature_filter_query_in),
+        ],
+        strict=False,
+    ):
+        plt.sca(ax)
+        x = df_plot["af"].to_numpy()
+        tot_mutations = df_plot.shape[0]
+        h, bin_edges = np.histogram(x, bins=bins)
+        bin_centers = (bin_edges[1:] + bin_edges[:-1]) / 2
+        plt.fill_between(
+            bin_centers,
+            -10,
+            h,
+            label=f"Median = {np.median(x):.1%}\nMean = {np.mean(x):.1%}",
+        )
+        plt.legend()
+        plt.xlim(0, 1)
+        plt.ylim(-1, ax.get_ylim()[1])
+        plt.xlabel("AF")
+        plt.title(f"{column}, total={tot_mutations:,}", fontsize=28)
+    plt.show()
+
+
+ZERO_TF_FILL = 1e-7
```

Comment:
constant - should either go in the top of the file, or better yet be a keyword argument in the function that uses it